src/Build/BackEnd/Components/SdkResolution/SdkResolverService.cs

```diff
@@ -319,18 +365,8 @@ private bool TryResolveSdkUsingSpecifiedResolvers(IList<SdkResolver> resolvers,
                 results.Add(result);
             }
 
-            foreach (SdkResult result in results)
-            {
-                LogWarnings(loggingContext, sdkReferenceLocation, result);
-
-                if (result.Errors != null)
-                {
-                    foreach (string error in result.Errors)
-                    {
-                        loggingContext.LogErrorFromText(subcategoryResourceName: null, errorCode: null, helpKeyword: null, file: new BuildEventFileInfo(sdkReferenceLocation), message: error);
-                    }
-                }
-            }
+            warnings = results.SelectMany(r => r.Warnings ?? Array.Empty<string>());
+            errors = results.SelectMany(r => r.Errors ?? Array.Empty<string>());
```

Comment:
Yes. We put all the errors together and all the warnings together. See https://github.com/dotnet/msbuild/blob/43eea93ed2b438aa24c10d6e1388d047de23cd3b/src/Build/BackEnd/Components/SdkResolution/SdkResolverService.cs#L225-L230, for instance.

Reply:
Since the order is relevant, this seems bad. Can we preserve the order?

Reply:
You mean interleave errors and warnings?

Reply:
Yes.

Reply:
I discussed that with baronfel two months ago, and we decided this was better.

Reply:
Can you explain the reasoning? Order is relevant so reordering the messages seems like an impediment to debugging to me.

Reply:
If we assume "error" means "this is why we failed to resolve the SDK," and "warning" means "here's something else you might want to change," then interspersing them means a user might see just one error first, followed by a warning or two then more errors, and assume that that error is the first one they should try to resolve, which is not necessarily true. In other words, if someone sees:

error1
warning1
warning2
error2
error3
warning3

Then because there's a gap between the first and second error, they'd treat them as separate and try to figure out why the default sdk resolver failed to resolve the SDK when that might be a complete red herring.

Second, (a weaker reason in my view) we don't want to show any errors if any of the SDK resolvers succeeds, again because it is expected that some will fail. That means we have to store the errors and display them later anyway. We're also only showing any errors if the user tells us to, whereas we always show the warnings. Each of these means we cannot just show the warnings and errors as they show up but rather have to keep track of them and log them later. If other things pop up in the interim, they may be logged immediately. That means we've already lost the found-a-bug-reported-a-bug connection; this just takes it one small step further.

Reply:
I do not find this convincing but we can try it.